pymbolic/mapper/__init__.py

```diff
@@ -491,24 +535,39 @@ def do_map(expr):
             else:
                 return self.rec(expr, *args, **kwargs)
 
-        return type(expr)(
-                tuple(do_map(ch) for ch in expr.children))
+        children = tuple(do_map(child) for child in expr.children)
```

Comment:
```suggestion
        children = tuple(None if child is None else self.rec(child) for child in expr.children)
```
(and delete `do_map`)

Reply:
Done